cosmwasm/contracts/v1/compute-tests/bench-contract/src/contract.rs

```diff
@@ -44,17 +57,87 @@ pub fn execute(
             bench_read_large_key_from_storage(deps, 2)
         }
         ExecuteMsg::BenchWriteLargeItemToStorage { .. } => bench_write_large_storage_key(deps, 2),
+        ExecuteMsg::CreateViewingKey {} => try_create_key(deps, env, info),
+        ExecuteMsg::SetViewingKey { key, .. } => try_set_key(deps, info, key),
     };
 
     Ok(Response::default())
 }
 
 #[entry_point]
-pub fn query(_deps: Deps, _env: Env, _msg: QueryMsg) -> StdResult<Binary> {
-    Ok(Binary::default())
+pub fn query(deps: Deps, _env: Env, msg: QueryMsg) -> StdResult<Binary> {
+    match msg {
+        QueryMsg::WithPermit { permit, query } => permit_queries(deps, permit, query),
+        _ => viewing_keys_queries(deps, msg),
+    }
 }
 
 #[entry_point]
 pub fn reply(_deps: DepsMut, _env: Env, _reply: Reply) -> StdResult<Response> {
     Ok(Response::default())
 }
+
+pub fn try_create_key(deps: DepsMut, env: Env, info: MessageInfo) -> StdResult<()> {
+    let _key = ViewingKey::create(
+        deps.storage,
+        &info,
+        &env,
+        info.sender.as_str(),
+        "some_entropy".as_bytes(),
+    );
+    Ok(())
+}
+
+pub fn try_set_key(deps: DepsMut, info: MessageInfo, key: String) -> StdResult<()> {
+    ViewingKey::set(deps.storage, info.sender.as_str(), key.as_str());
+    Ok(())
+}
+
+fn permit_queries(deps: Deps, permit: Permit, query: QueryWithPermit) -> Result<Binary, StdError> {
```

Comment:
In my opinion, the name is misleading here, I would suggest changing to the following:
```suggestion
fn query_with_permit(deps: Deps, permit: Permit, query: QueryWithPermit) -> Result<Binary, StdError> {
```